crates/matrix-sdk/src/room/mod.rs

```diff
@@ -397,19 +398,28 @@ impl Room {
             None
         };
 
+        #[cfg(feature = "e2e-encryption")]
+        let enable_share_history_on_invite = self.client.inner.enable_share_history_on_invite;
+
+        #[cfg(not(feature = "e2e-encryption"))]
+        let enable_share_history_on_invite = false;
+
+        debug!(
+            ?prev_room_state,
+            inviter=?inviter.as_ref().map(|room_member| room_member.user_id()),
+            enable_share_history_on_invite,
+            "Joining room",
+        );
+
         self.client.join_room_by_id(self.room_id()).await?;
 
         #[cfg(feature = "e2e-encryption")]
```

Comment:
We could remove this if we were OK with the extra runtime check?

Reply:
unfortunately that leads to a build error due to the absence of `shared_room_history::maybe_accept_key_bundle`.